worker/src/lib/HeliconeProxyRequest/forwarder.ts

```diff
@@ -21,6 +21,18 @@ export async function proxyForwarder(
   ctx: ExecutionContext,
   provider: Provider
 ): Promise<Response> {
+  let organizationId = null;
+  const { data: auth, error: authError } = await request.auth();
```

Comment:
This is doing the auth stuff too early. We do not want to add latency or authenticate users to just proxy requests. This should only be done after the shouldReadFromCache resolves to be true. Then we do this and check if they are authenticated. If they are not, we should just proxy normally without using the cache.

Reply:
Resolved this with [4199e61](https://github.com/Helicone/helicone/pull/1174/commits/4199e6147ce71ce739834f0370a2f017877a4247) by checking if shouldReadFromCache is present.